frame/staking/src/migrations.rs

```diff
@@ -18,6 +18,59 @@
 
 use super::*;
 
+pub mod v8 {
+	use frame_election_provider_support::SortedListProvider;
+	use frame_support::traits::Get;
+
+	use crate::{Config, Nominators, Pallet, StorageVersion, Weight};
+
+	#[cfg(feature = "try-runtime")]
+	pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+		frame_support::ensure!(
+			StorageVersion::<T>::get() == crate::Releases::V7_0_0,
+			"must upgrade linearly"
+		);
+		frame_support::ensure!(
+			T::SortedListProvider::iter().count() == 0,
+			"voter list already exists"
+		);
+		crate::log!(info, "👜 staking bags-list migration passes PRE migrate checks ✅",);
+		Ok(())
+	}
+
+	/// Migration to sorted [`SortedListProvider`].
+	pub fn migrate<T: Config>() -> Weight {
+		if StorageVersion::<T>::get() == crate::Releases::V7_0_0 {
```

Comment:
we have to be careful that in polkadot/kusama as long as there is the old migration running the storage version is always overwritten to the latest version.

https://github.com/paritytech/polkadot/pull/3731

Reply:
Can you clarify? From what I understand this should be fine since this is a storage item only for the staking pallet and would not be modified anywhere else.

Reply:
ah indeed my bad, it is not part of the FRAME storage version stuff. Then yes there shouldn't be any conflict